### PR TITLE
[SMALLFIX]  More improvements to async persist

### PR DESCRIPTION
**Pull request by peisun1115**
1. Do not do heavy operation in the heartbeat 
2. Increase the threadpool size from 2 to 64 (persist is an IO bound synchronous hadoop file operation. 2 threads are far from enough to achieve reasonable throughput when the file size is not huge).
Note that 64 is not necessarily sufficient to achieve the maximum throughput we can get with one worker.  I just picked a small number to be conservative for now since we are using TThreadPoolServer which uses lots of threads already.

Here are the logs (format: timestamp number_of_files_persisted)

Before this change:
2016-06-08 16:30:34,182 8498
2016-06-08 16:30:35,181 178
2016-06-08 16:30:36,181 187
2016-06-08 16:30:37,181 191
2016-06-08 16:30:38,183 200
2016-06-08 16:30:39,184 203
2016-06-08 16:30:40,183 91
**2016-06-08 16:29:46,367 112598
2016-06-08 16:19:18,260 2926**

After this change but keep the number of threads the same (i.e .2):
After: With 2 threads.
2016-06-08 17:43:05,216 165
2016-06-08 17:43:06,216 181
2016-06-08 17:43:07,216 179
2016-06-08 17:43:08,216 103
2016-06-08 17:43:09,216 133
2016-06-08 17:43:10,216 168
2016-06-08 17:43:11,216 170
2016-06-08 17:43:12,216 162
2016-06-08 17:43:13,216 166
2016-06-08 17:43:14,216 157
2016-06-08 17:43:15,217 171
..... (more)

After this change with 64 threads:
....
016-06-08 18:40:19,297 515
2016-06-08 18:40:20,298 809
2016-06-08 18:40:21,299 937
2016-06-08 18:40:22,298 827
2016-06-08 18:40:23,298 862
2016-06-08 18:40:24,298 957
2016-06-08 18:40:25,299 908
2016-06-08 18:40:26,298 912
2016-06-08 18:40:27,298 636
2016-06-08 18:40:28,298 869
2016-06-08 18:40:29,298 934
2016-06-08 18:40:30,298 850
2016-06-08 18:40:31,298 923
2016-06-08 18:40:32,298 756
2016-06-08 18:40:33,298 419
2016-06-08 18:40:34,298 859
2016-06-08 18:40:35,299 935
2016-06-08 18:40:36,299 902
2016-06-08 18:40:37,299 775
....

### CODE REVIEW

**Comment by peisun1115** [comment]
@apc999 @calvinjia

File: core/server/src/main/java/alluxio/worker/file/FileWorkerMasterSyncExecutor.java [hunk 1]
```diff
@@ -147,11 +137,19 @@ public FilePersister(FileDataManager fileDataManager, long fileId, List<Long> bl
 
     @Override
     public void run() {
-      try {
+      if (mFileDataManager.needPersistence(mFileId)) {
+        // lock all the blocks of the file to prevent eviction
```
Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/file/FileWorkerMasterSyncExecutor.java]: Now that we're doing this asynchronously, the file could be evicted before the executor schedules this thread. Do we need to pin the file on the master side before sending the persist command? Do we guarantee that an async persist file won't be evicted before being persisted?

Comment (aaudiber) [inline core/server/src/main/java/alluxio/worker/file/FileWorkerMasterSyncExecutor.java]: Never mind, we didn't make this guarantee before anyway. When we add the guarantee we will probably do it on the writing to memory path and it won't matter what we do here.

**Comment by apc999** [comment]
+1 LGTM. thanks for the great improvement!

**Comment by aaudiber** [comment]
LGTM, thanks for fixing this!

**Pull request merged**